src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -248,3 +250,107 @@ def _try_fetch(self) -> Iterable[Mount]:
             f"SELECT * FROM {escape_sql_identifier(self._schema)}.{escape_sql_identifier(self._table)}"
         ):
             yield Mount(*row)
+
+
+@dataclass
+class TableInMount:
+    is_delta: bool
+    is_parquet: bool
+    is_partitioned: bool
+
+
+class TablesInMounts(CrawlerBase[Table]):
+
+    def __init__(
+        self,
+        backend: SqlBackend,
+        ws: WorkspaceClient,
+        inventory_database: str,
+        mc: Mounts,
+        include_mounts: list[str] | None = None,
+    ):
+        super().__init__(backend, "hive_metastore", inventory_database, "tables", Table)
+        self._dbutils = ws.dbutils
+        self._mc = mc
+        self._include_mounts = include_mounts
+
+    def snapshot(self) -> list[Table]:
+        """
+        Takes a snapshot of tables in the specified catalog and database.
+
+        Returns:
+            list[Table]: A list of Table objects representing the snapshot of tables.
+        """
+        return self._snapshot(partial(self._try_load), partial(self._crawl))
+
+    def _try_load(self) -> Iterable[Table]:
+        """Tries to load table information from the database or throws TABLE_OR_VIEW_NOT_FOUND error"""
+        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
+            yield Table(*row)
+
+    def _crawl(self):
+        all_mounts = self._mc.snapshot()
+        all_tables = []
+        for mount in all_mounts:
+            if self._include_mounts and mount.name not in self._include_mounts:
+                continue
+            table_paths = self._find_delta_log_folders(mount.name)
+            for path, entry in table_paths.items():
+                table = Table(
+                    catalog="hive_metastore",
+                    database="tables_in_mounts",
+                    name=self._get_table_name(path),
+                    object_type="EXTERNAL",
+                    table_format="DELTA" if entry.is_delta else "PARQUET",
+                    location=path,
+                )
+                all_tables.append(table)
+        return all_tables
+
+    def _find_delta_log_folders(self, root_dir, delta_log_folders=None) -> dict:
+        if delta_log_folders is None:
+            delta_log_folders = {}
+
+        entries = self._dbutils.fs.ls(root_dir)
+        for entry in entries:
+            if self._is_irrelevant(entry.name):
+                continue
+
+            path = self._get_absolute_directory(entry.path)
```

Comment:
```suggestion
            path = os.path.dirname(entry.path)
```
standard lib has it all